tests/ctags/var-and-return-type.cpp

```diff
@@ -0,0 +1,40 @@
+
+const volatile unsigned int func1();
+const volatile unsigned int var1 = 0;
+
+struct type1 {
+	unsigned int memb1;
+	struct type1 *next;
+};
+
+const struct type1 func2();
+const struct type1 var2 = { 0, 0 };
+
+typedef type1 type1_t;
+
+const type1_t func3();
+const type1_t var3 = { 0, 0 };
+
+struct type1 func4();
+struct type1 var4 = { 0, 0 };
+
+type1_t func5();
+type1_t var5 = { 0, 0 };
+
```

Comment:
> what about testing  `type1_t var{};` or `type1_t var{ 0, 0 };`

Nothing to do with reporting the var type, it's "just" another mean of initializing.  So different issue, possibly different test case.

> what about testing `auto func()->type1_t` C++11 syntax 

This is more related, but still quite outside the scope here, as AFAIK `->` syntax is not supported at all yet. (OT, WTF is about this syntax?  why add a new one?)